### PR TITLE
added get connections

### PR DESCRIPTION
**Pull request by rauljareno**

### CODE REVIEW

File: lib/iden3comm/data/data_sources/storage_connection_data_source.dart [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+import 'package:polygonid_flutter_sdk/sdk/di/injector.dart';
+import 'package:sembast/sembast.dart';
+import 'package:sembast/utils/sembast_import_export.dart';
+
+/// [StoreRef] wrapper
+/// Delegates all call to [ConnectionStoreRefWrapper._store]
+/// Needed for UT for mocking extension methods
+@injectable
+class ConnectionStoreRefWrapper {
+  final Map<String, StoreRef<String, Map<String, Object?>>> _store;
```
Comment (Plumillon) [inline lib/iden3comm/data/data_sources/storage_connection_data_source.dart:15]: Only inject the source of this DataSource, not all Stores (see my comment below).

File: lib/iden3comm/domain/use_cases/connection/get_connections_use_case.dart [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  Future<List<ConnectionEntity>> execute(
+      {required GetConnectionsParam param}) async {
+    // if profileNonce is zero, return all profiles claims,
+    // if profileNonce > 0 then return only claims from that profile
+    if (param.profileNonce >= 0) {
+      if (param.profileNonce > 0) {
+        String did = await _getCurrentEnvDidIdentifierUseCase.execute(
+            param: GetCurrentEnvDidIdentifierParam(
+                privateKey: param.privateKey,
+                profileNonce: param.profileNonce));
```
Comment (Plumillon) [inline lib/iden3comm/domain/use_cases/connection/get_connections_use_case.dart:40]: Mutualize that (minor).

**Review by Plumillon — CHANGES_REQUESTED** [review]
Nice!

File: lib/iden3comm/data/data_sources/storage_connection_data_source.dart (hunk 1 above)
Comment (rauljareno) [reply lib/iden3comm/data/data_sources/storage_connection_data_source.dart:15]: will do in another PR

**Review by Plumillon — APPROVED** [review]

**Pull request merged**